<!DOCTYPE html>
<html>
<head>
<title>02_quality.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="galaxy-for-virologist-training-exercise-2-quality-control-and-trimming">Galaxy for virologist training Exercise 2: Quality control and trimming</h1>
<p>Despite the improvement of sequencing methods, there is no error-free technique. A correct measuring of the sequencing quality is essential for identifying problems in the sequencing, thus, this must be the first step in every sequencing analysis. Once the quality control is finished, it's important to remove those low quality reads, or short reads, for which a trimming step is mandatory. After the trimming step it is recommended to perform a new quality control step to be sure that trimming worked.</p>
<h2 id="1-illumina-quality-control-and-trimming">1. Illumina Quality control and trimming</h2>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th><strong>Title</strong></th>
<th>Pre-processing</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Training dataset:</strong></td>
<td>PRJEB43037 - In August 2020, an outbreak of West Nile Virus affected 71 people with meningoencephalitis in Andalusia and 6 more cases in Extremadura (south-west of Spain), causing a total of eight deaths. The virus belonged to the lineage 1 and was relatively similar to previous outbreaks occurred in the Mediterranean region. Here, we present a detailed analysis of the outbreak, including an extensive phylogenetic study. This is one of the outbreak samples.</td>
</tr>
<tr>
<td><strong>Questions:</strong></td>
<td><ul><li>How do I check whether my Illumina data was correctly sequenced?</li><li>How can I improve the quality of my data?</li></ul></td>
</tr>
<tr>
<td><strong>Objectives</strong>:</td>
<td><ul><li>Perform a quality control in raw Illumina reads</li><li>Perform a quality trimming in raw Illumina reads</li><li>Perform a quality control in trimmed Illumina reads</li></ul></td>
</tr>
<tr>
<td><strong>Estimated time</strong>:</td>
<td>25 min</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<h3 id="11-quality-control">1.1. Quality control</h3>
<h4 id="111-upload-data">1.1.1. Upload data</h4>
<p>To run the quality control over the samples, follow these steps:</p>
<ol>
<li><a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#2-galaxys-history">Create a new history, as we explained yesterday</a> named <strong>Illumina preprocessing</strong></li>
<li><a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#3-loading-data">Upload data as seen yesterday</a>, copy and paste the following URLs:</li>
</ol>
<pre class="hljs"><code><div>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_2.fastq.gz
</div></code></pre>
<ol start="3">
<li>Add some tags to the files. <em><strong>It is mandatory that the tag starts with <code>#</code> to be propagated to the processes</strong></em>.</li>
</ol>
<p><img src="images/add_tag1.png" alt="add_tag1" width="250"><img src="images/add_tag2.png" alt="add_tag2" width="250"><img src="images/add_tag3.png" alt="add_tag3" width="250"></p></p>
<h4 id="112-run-fastqc">1.1.2. Run FastQC</h4>
<ol>
<li>Search for the <strong>fastqc</strong> tool</li>
<li>Select <strong>FastQC Read Quality reports</strong> and set the following parameters:</li>
<li>Select multiple file data set in Raw read data from your current history</li>
<li>Select the two datasets</li>
<li>Then go down and select <strong>Run tool</strong></li>
</ol>
<p align="center"><img src="images/fastqc_run.png" alt="fastqc_run" width="900"></p>
<p>To see the results we are going to open the jobs with <strong>Web page</strong> in their name for both data 1 and data 2.</p>
<p align="center"><img src="images/fastqc_results_r1r2.png" alt="fastqc_results_r1r2" width="1000"></p>
<p>Here, you can see the number of reads in each file, the maximum and minimum length of all reads in the sample, and the quality plots for both R1 and R2. They look quite good, but we are going to run trimming over the samples.</p>
<details>
<summary>How many reads do the samples have?</summary>
<br>
265989
</details>
<p><strong>First question</strong></p>
<details>
<summary>How do I check whether my Illumina data was correctly sequenced?</summary>
<br>
Using FastQC
</details>
<h3 id="12-trimming">1.2. Trimming</h3>
<p>Once we have performed the quality control, we have to perform the quality and read length trimming:</p>
<h4 id="121-run-fastp">1.2.1. Run Fastp</h4>
<p>1.Search for <strong>fastp</strong> in the tools</p>
<p>2.Then select <strong>fastp - fast all-in-one preprocessing for FASTQ files</strong></p>
<p>-Select custom parameters:</p>
<pre><code>3.Single-end or paired reads &gt; Paired

    4.Input 1 &gt; Browse datasets (right folder icon) &gt; Select ERR5310322_1.fastq.gz

    5.Input 2 &gt; Browse datasets &gt; Select ERR5310322_2.fastq.gz

6.Display Filter Options

    -Quality Filtering options

        7.Qualified Quality Phred = 30

        8.Unqualified percent limit = 10

    -Length Filtering Options

        9.Length required = 50

10.Read modification options

    11.PoliX tail trimming &gt; Enable polyX tail trimming

    -Per read cutting by quality options

        12.Cut by quality in front (5') &gt; Yes

        13.Cut by quality in tail (3') &gt; Yes

        14.Cutting mean quality = 30
</code></pre>
<p>15.Finally, click on <strong>Run tool</strong></p>
<p align="center"><img src="images/fastp_1.png" alt="fastp_1" width="900"></p>
<p align="center"><img src="images/fastp_2.png" alt="fastp_2" width="900"></p>
<p align="center"><img src="images/fastp_3.png" alt="fastp_3" width="900"></p>
<p>To see the trimming stats, have a look at the <strong>fastp on data 2 and data 1: HTML report</strong> file. You should see something like that.</p>
<p align="center"><img src="images/fastp_results.png" alt="fastp_results" width="700"></p>
<details>
<summary>How many reads have we lost?</summary>
<br>
98664 reads
</details>
<h4 id="122-other-trimming-tools-trimmomatic">1.2.2. Other trimming tools: Trimmomatic</h4>
<p>1.Search for <strong>trimmomatic</strong> in the tools</p>
<p>2.Select <strong>Trimmomatic flexible read trimming tool for Illumina NGS data</strong></p>
<p>-Select custom parameters:</p>
<pre><code>3.Single-end or paired-end reads? = Paired-end (two separated files)

4.Input FASTQ file (R1/first of pair) = ERR5310322_1.fastq.gz

5.Input FASTQ file (R2/second of pair) = ERR5310322_2.fastq.gz

6.Average quality required = 30

7.Insert Trimmomatic Operation:

    8.Select Trimmomatic operation to perform: **MINLEN**

    9.Minimum length of reads to be kept = 50
</code></pre>
<p>10.Select <strong>Run tool</strong></p>
<p align="center"><img src="images/trimmomatic_1.png" alt="trimmomatic_1" width="900"></p>
<p align="center"><img src="images/trimmomatic_2.png" alt="trimmomatic_2" width="900"></p>
<p>Trimmomatic does not perform statistics over trimmed reads, so we need to perform FastQC again over the Trimmomatic results.</p>
<details>
<summary>Try to do it on your own.</summary>
<br>
<p align="center"><img src="images/fastqc_trimmomatic.png" alt="fastqc_trimmomatic" width="900"></p>
<p align="center"><img src="images/fasqc_trimming_res.png" alt="fasqc_trimming_res" width="900"></p>
</details>
<p><strong>Second question</strong></p>
<details>
<summary>How can I improve the quality of my data?</summary>
<br>
Using a trimming software, such as fastp or trimmomatic.
</details>
<ul>
<li>This hands-on history URL: <a href="https://usegalaxy.eu/u/svarona/h/illumina-preprocessing">https://usegalaxy.eu/u/svarona/h/illumina-preprocessing</a></li>
</ul>
<h2 id="2-nanopore-quality-control-and-trimming">2. Nanopore Quality control and trimming</h2>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th><strong>Title</strong></th>
<th>Galaxy</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Training dataset:</strong></td>
<td>The data we are going to manage corresponds to Nanopore amplicon sequencing data using ARTIC network primers por SARS-CoV-2 genome. From the Fast5 files generated by the ONT software, we are going to select the pass reads, so they are already filtered by quality.</td>
</tr>
<tr>
<td><strong>Questions:</strong></td>
<td><ul><li>How do I know if my Nanopore data was correctly sequenced?</li></ul></td>
</tr>
<tr>
<td><strong>Objectives</strong>:</td>
<td><ul><li>Perform a quality control in raw Illumina reads</li><li>Perform a quality trimming in raw Nanopore reads</li><li>Perform a quality control in trimmed Nanopore reads</li></ul></td>
</tr>
<tr>
<td><strong>Estimated time</strong>:</td>
<td>15 min</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<h3 id="21-quality-control">2.1. Quality control</h3>
<p>To run the quality control over the samples, follow these steps:</p>
<ol>
<li><a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#2-galaxys-history">Create a new history has explained yesterday</a> named <strong>Nanopore quality</strong></li>
<li><a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#3-loading-data">Upload data as seen yesterday</a>, copy and paste the following URLs:</li>
</ol>
<pre class="hljs"><code><div>https://raw.githubusercontent.com/nf-core/<span class="hljs-built_in">test</span>-datasets/viralrecon/nanopore/minion/fastq_pass/barcode01/FAO93606_pass_barcode01_7650855b_0.fastq
https://raw.githubusercontent.com/nf-core/<span class="hljs-built_in">test</span>-datasets/viralrecon/nanopore/minion/fastq_pass/barcode01/FAO93606_pass_barcode01_7650855b_1.fastq
https://raw.githubusercontent.com/nf-core/<span class="hljs-built_in">test</span>-datasets/viralrecon/nanopore/minion/fastq_pass/barcode01/FAO93606_pass_barcode01_7650855b_2.fastq
</div></code></pre>
<h4 id="211-pycoqc">2.1.1. PycoQC</h4>
<p>To use PycoQC we need to use the <code>sequencing_summary.txt</code> provided by de Nanopore sequencing machine.</p>
<p><a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#3-loading-data">Upload data as seen yesterday</a>, copy and paste the following URL:</p>
<pre class="hljs"><code><div>https://raw.githubusercontent.com/nf-core/<span class="hljs-built_in">test</span>-datasets/viralrecon/nanopore/minion/sequencing_summary.txt
</div></code></pre>
<ol>
<li>Search for the <strong>Pycoqc</strong> tool</li>
<li>Select <strong>Pycoqc quality control for Nanopore sequencing data</strong></li>
<li>In <em>A sequencing_summary file</em>: Select the <code>sequencing_summary.txt</code> we just uploaded</li>
<li>Select <strong>Run tool</strong></li>
</ol>
<p align="center"><img src="images/pycoqc.png" alt="pycoqc_run" width="900"></p>
<p>Then inspect the resulting PycoQC HTML Report:</p>
<p align="center"><img src="images/pycoqc_output_1.png" alt="pycoqc_output_1" width="900"></p>
<p><strong>Question</strong></p>
<details>
<summary>How many reads do the samples have?</summary>
<br>
3K reads
</details>
<details>
<summary>Do you understand all the plots?</summary>
<br>
<b>Basecalled reads length</b>
<p align="center"><img src="images/pycoqc_output_2.png" alt="pycoqc_output_2" width="900"></p>
This plot shows the distribution of fragment sizes in the file that was analyzed. Long reads have a variable length and this will show the relative amounts of each different size of sequence fragment. In this example, the distribution of read length is quite dispersed with a minimum read length for the passed reads around 150 and a maximum length ~5000bp. However, most of the reads are about 500 nt length, as expected by the amplicon experiment.
<br>
<br>
<b>Basecalled reads PHRED quality:</b>
<p align="center"><img src="images/pycoqc_output_3.png" alt="pycoqc_output_3" width="900"></p>
This plot shows the distribution of the Qscores (Q) for each read. This score aims to give a global quality score for each read. The exact definition of Qscores is: the average per-base error probability, expressed on the log (Phred) scale. In case of Nanopore data, the distribution is generally centered around 10 or 12. For old runs, the distribution can be lower, as basecalling models are less precise than recent models. In our case, the median read Qscore is 13, which means that this run has good quality.
<br>
<br>
<b>Basecalled reads length vs reads PHRED quality:</b>
<p align="center"><img src="images/pycoqc_output_4.png" alt="pycoqc_output_4" width="900"></p>
This representation give a 2D visualisation of read Qscore according to the length.
<br>
<br>
<b>Output over experiment time:</b>
<p align="center"><img src="images/pycoqc_output_5.png" alt="pycoqc_output_5" width="900"></p>
This representation gives information about sequenced reads over the time for a single run. We can see that the production of reads is decreasing over time, which can be due to the sequencing of mosth of the genetic material, the saturation of pores and/or the degradation of the marial and/or pores. In this example, the “Cummulative” plot area (light blue) indicates that 50% of all reads and almost 50% of all bases were produced in the first 3h of the 8h experiment. We can see that from 6 to 8h of the experiment, only 200 reads were yield, which means that we could have ended the experiment 2h before.
<br>
<br>
<b>Read length over experiment time:</b>
<p align="center"><img src="images/pycoqc_output_6.png" alt="pycoqc_output_6" width="900"></p>
The read length over experiment time should be stable. It can slightly increase over the time as short fragments tend to be over-sequenced at the beginning and are less present over the time. In this case, as almost all the fragments have same length, the plot is really constant over time.
<br>
<br>
 <b>Read quality over experiment time:</b>
<p align="center"><img src="images/pycoqc_output_7.png" alt="pycoqc_output_7" width="900"></p>
The read quality over experiment time should be stable too, but usually it slightly decrease over the time as pores get saturated or degraded. In this case, we can see a clear decrease of sequencing quality over experiment time, but it mantains between the good quality values and this can be fixed with further post processing of the reads.
<br>
<br>
<p><b>Number of reads per barcode</b>:</p>
<p align="center"><img src="images/pycoqc_output_9.png" alt="pycoqc_output_9" width="900"></p>
This plot shows the number of reads per barcode, which means de number of reads per sample to be demultiplexed. In a goog experiment, all the barcodes should have the same number of reads. In this training we only used reads from barcode01 sample but we can see that barcode08 couldn't be correctly sequenced.
<br>
<br>
<b>Channel activity over time:</b>
<p align="center"><img src="images/pycoqc_output_8.png" alt="pycoqc_output_8" width="900"></p>
It gives an overview of available pores, pore usage during the experiment, inactive pores and shows if the loading of the flow cell is good (almost all pores are used). In this case, the vast majority of channels/pores are inactive (white) after the 6h of experiment, so the run should have been dinished at that time. You would hope for a plot that it is dark near the X-axis, and with higher Y-values (increasing time) doesn’t get too light/white. Depending if you chose “Reads” or “Bases” on the left the colour indicates either number of bases or reads per time interval.
</details>
<details>
<summary>How do I check whether my Nanopore data was correctly sequenced?</summary>
<br>
Using NanoPlot or PycoQC and having a look to the statistic values.
</details>
<h3 id="22-trimming">2.2. Trimming</h3>
<p>When Nanopore reads are being sequenced, the MinKnown software splits Fast5 reads into quality pass and quality fail. As we will select only Fast5 pass reads, we won't need to perform a quality trimming, so even if we see that the reads have a bad Phred score, we know that the ONT software considered the reads as &quot;good quality&quot;.</p>
<p>Then we will only be performing a read length trimming. As we are using amplicon sequencing data, we won't be expecting reads smaller than 400 nucleotides, nor higher than 600, which would obviously correspond to chimeric reads.</p>
<h4 id="221-artic">2.2.1. Artic</h4>
<ol>
<li>Search for <strong>artic</strong> tool</li>
<li>Select <strong>ARTIC guppyplex Filter Nanopore reads by read length and (optionally) quality</strong></li>
<li>Structure of your input data: Multiple input datasets per sample</li>
<li>While pressing the <em>Ctrl</em> key, select the three samples</li>
<li>Remove reads longer than = 600</li>
<li>Remove reads shorter than = 300</li>
<li>Do not filter on quality score (speeds up processing) = Yes (we had already select pass reads)</li>
</ol>
<p align="center"><img src="images/artic_filter.png" alt="nanofilt_run" width="900"></p>
<h4 id="222-nanoplot">2.2.2. Nanoplot</h4>
<p>Now we are going to run NanoPlot on filtered data:</p>
<ol>
<li>Search for the <strong>Nanoplot</strong> tool and select <strong>NanoPlot Plotting suite for Oxford Nanopore sequencing data and alignments</strong></li>
<li>Run the tool as follows:
<ul>
<li>In the <em>files</em> part, select ARTIC output file.</li>
<li>Display <strong>Options for customizing the plots created</strong>:
<ul>
<li><strong>Specify the bivariate format of the plots</strong> &gt; <em>Select all</em></li>
<li><strong>Show the N50 mark in the read length histogram</strong> &gt; <em>Yes</em></li>
</ul>
</li>
<li>Select <strong>Execute</strong></li>
</ul>
</li>
</ol>
<p align="center"><img src="images/nanoplot_filtered_1.png" alt="nanoplot_filtered_1" width="900"></p>
<p align="center"><img src="images/nanoplot_filtered_2.png" alt="nanoplot_filtered_2" width="900"></p>
<p><strong>Questions</strong></p>
<details>
<summary>Did our data length and quality improve?</summary>
<br>
Yes, now we hace reads in the length and quality specified.
</details>
<details>
<summary>How many reads did we lost during trimming step?</summary>
<br>
137 reads
</details>
<ul>
<li>This hands-on history URL: <a href="https://usegalaxy.eu/u/s.varona/h/nanopore-quality">https://usegalaxy.eu/u/svarona/h/nanopore-quality</a></li>
</ul>
<blockquote>
<p><strong><em>NOTE:</em></strong>  We can't use nanofilt because it is not installed in Galaxy</p>
</blockquote>

</body>
</html>
