<!DOCTYPE html>
<html>
<head>
<title>05_mapping.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="galaxy-for-virologist-training-exercise-5-illumina-mapping-101">Galaxy for virologist training Exercise 5: Illumina Mapping 101</h1>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th><strong>Title</strong></th>
<th>Galaxy</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Training dataset:</strong></td>
<td>PRJEB43037 - In August 2020, an outbreak of West Nile Virus affected 71 people with meningoencephalitis in Andalusia and 6 more cases in Extremadura (south-west of Spain), causing a total of eight deaths. The virus belonged to the lineage 1 and was relatively similar to previous outbreaks occurred in the Mediterranean region. Here we present a detailed analysis of the outbreak, including an extensive phylogenetic study. This is one of the outbreak samples.</td>
</tr>
<tr>
<td><strong>Questions:</strong></td>
<td><ul><li>What is mapping?</li><li>What is a BAM file?</li><li>Which metrics are important to check after mapping?</ul></td>
</tr>
<tr>
<td><strong>Objectives</strong>:</td>
<td><ul><li>Understand the concept of mapping</li><li>Learn how to interpret mapping metrics</li><li>Learn how to visualize mapping results</li></ul></td>
</tr>
<tr>
<td><strong>Estimated time</strong>:</td>
<td>40 min</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<h2 id="1-description">1. Description</h2>
<p>One of the most common experiments using massive sequencing are re-sequencing experiments. This type of experiments sequence already known microorganisms, with the goal to discover variation between an already assembled and known reference, and our reads. Mapping is a mandatory step for this kind of experiments, where we need to sort all the short sequences (reads) we have in our fastq file, lacking any genomic context.
After the mapping step, we will transform our fastq file into a bam file that contains information about where a read came from, meaning we are going to have the coordinates where each read is placed inside our reference genome.</p>
<h2 id="2-upload-data-to-galaxy">2. Upload data to galaxy</h2>
<h3 id="training-dataset">Training dataset</h3>
<ul>
<li>Experiment info: PRJEB43037, WGS, Illumina MiSeq, paired-end</li>
<li>Fastq R1: <a href="https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_1.fastq.gz">ERR5310322_1</a> - url : <code>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_1.fastq.gz</code></li>
<li>Fastq R2: <a href="https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_2.fastq.gz">ERR5310322_2</a>  url : <code>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_2.fastq.gz</code></li>
<li>Reference genome NC_009942.1: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/875/385/GCF_000875385.1_ViralProj30293/GCF_000875385.1_ViralProj30293_genomic.fna.gz">fasta</a> -- <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/875/385/GCF_000875385.1_ViralProj30293/GCF_000875385.1_ViralProj30293_genomic.gff.gz">gff</a></li>
</ul>
<h3 id="create-new-history">Create new history</h3>
<ul>
<li>Click the <code>+</code> icon at the top of the history panel and create a new history with the name <code>mapping 101 tutorial</code> as explained <a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#2-galaxys-history">here</a></li>
</ul>
<h3 id="upload-data">Upload data</h3>
<ul>
<li>Import and rename the read files <code>ERR5310322_1</code> and <code>ERR5310322_2</code>
<ol>
<li>Click in upload data.</li>
<li>Click in paste/fetch data</li>
<li>Copy url for fastq R1 (select and Ctrl+C) and paste (Ctrl+V).</li>
<li>Click in Start.</li>
<li>Wait until the job finishes (green in history)</li>
<li>Do the same for fastq R2.</li>
</ol>
</li>
</ul>
<p align="center"><img src="images/upload_data_mapping.png" alt="Upload data mapping" width="900"></p>
<ul>
<li>Rename R1 and R2 files.
<ol>
<li>Click in ✏️ in the history for <code>ERR5310322_1.fastq.gz</code></li>
<li>Change the name to <code>ERR5310322_1</code></li>
<li>Do the same for R2.</li>
</ol>
</li>
</ul>
<p align="center"><img src="images/changename1.png" alt="Change name 1" width="900"></p>    
<ul>
<li>Import the reference genome:</li>
</ul>
<pre class="hljs"><code><div>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/875/385/GCF_000875385.1_ViralProj30293/GCF_000875385.1_ViralProj30293_genomic.fna.gz
</div></code></pre>
<ul>
<li>Rename the reference genome and gff file.
<ol>
<li>Click the ✏️ for the reference file in the history.</li>
<li>Change the name to <code>NC_009942.1</code></li>
</ol>
</li>
</ul>
<p align="center"><img src="images/changename2.png" alt="Change name 2" width="900"></p>    
<h3 id="map-reads-using-bowtie2">Map reads using Bowtie2</h3>
<ol>
<li>Search bowtie2 software in the search tools box on the left.</li>
</ol>
<p align="center"><img src="images/search_bowtie2.png" alt="Search bowtie" width="400"></p>   
<ol start="2">
<li>
<p>Set bowtie2 parameters:</p>
<ul>
<li>Is this single or paired library: paired.</li>
<li>FASTA/Q file #1 : ERR5310322_1</li>
<li>FASTA/Q file #2 : ERR5310322_2</li>
<li>Will you select a reference genome from your history or use a built-in index? : Use a genome from the history and build index.</li>
<li>Do you want to use presets? : Very sensitive local. This setting will hugly affect the mapping results, depending on the dataset/experiment must be tweaked (read <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">bowtie2 manual</a>)</li>
<li>Save the bowtie2 mapping statistics to the history</li>
</ul>
</li>
</ol>
<p align="center"><img src="images/bowtie2params1.png" alt="Search bowtie" width="900"></p>   
<p align="center"><img src="images/bowtie2params2.png" alt="Search bowtie" width="900"></p>   
<ol start="3">
<li>Click execute and wait.</li>
</ol>
<h3 id="visualize-bam-file-and-calculate-metrics">Visualize bam file and calculate metrics</h3>
<ol>
<li>Click the 👁️ icon in the Bowtie2 aligments in history.</li>
</ol>
<p align="center"><img src="images/view_bamfile.png" alt="view bam file" width="200"></p>
<ol start="2">
<li>
<p>Interpret the columns in the bam format according to the theory from class.</p>
</li>
<li>
<p>Visualize mapping metrics</p>
<ul>
<li>Click on the eye icon on Bowtie2 mapping stats history.</li>
</ul>
 <details>
 <summary> Which is the mapping rate?</summary>
 <br>
 93.44%
 </details>
</li>
<li>
<p>Calculate depth of coverage metrics using picard collectWGSMetrics.</p>
<ul>
<li>Search collectwgsmetrics on the search tool box.</li>
<li>Select SAM/BAM dataset or dataset collection: Bowtie2 alignments</li>
<li>Load reference genome from: History and select reference genome fasta file.</li>
<li>Treat bases with coverage exceeding this value as if they had coverage at this value: 3000</li>
</ul>
</li>
</ol>
<p align="center"><img src="images/collect_metrics.png" alt="view bam file" width="800"></p>
<ol start="5">
<li>
<p>Click execute and wait.</p>
 <details>
 <summary> Which is mean depth of coverage?</summary>
 <br>
 2805
 </details>
 <details>
 <summary> Which is genome coverage > 10x?</summary>
 <br>
 0.961193
 </details>
</li>
</ol>
<h3 id="visualize-bam-file-using-igv">Visualize bam file using IGV</h3>
<p>In order to visualize our mapping we will use IGV (Integrative Genomics Viewer). This is an open source, freely available and lightweight visualization tool that enables intuitive real-time exploration of diverse, large-scale genomic data sets on standard desktop computers. It supports flexible integration of a wide range of genomic data types including aligned sequence reads, mutations, copy number, RNA interference screens, gene expression, methylation and genomic annotations.</p>
<p>Navigation through a data set is similar to that of Google Maps, allowing the user to zoom and pan seamlessly across the genome at any level of detail, from whole genome to base pair. Data sets can be loaded from either local or remote sources, including cloud-based resources, enabling investigators to view their own genomic data sets alongside publicly available data.</p>
<ol>
<li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a></li>
<li>Launch IGV on your computer</li>
<li>Expand the param-file output of Bowtie2 tool</li>
<li>Click on the local in display with IGV to load the reads into the IGV browser</li>
<li><a href="https://training.galaxyproject.org/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html">Here</a> you have a galaxy training document for IGV usage.</li>
</ol>
<blockquote>
<p>This history is available at: https://usegalaxy.eu/u/smonzon/h/mapping-101-tutorial</p>
</blockquote>

</body>
</html>
