<!DOCTYPE html>
<html>
<head>
<title>06_mapping_nanopore.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="galaxy-for-virologist-training-exercise-4-nanopore-mapping-101">Galaxy for virologist training Exercise 4: Nanopore mapping 101</h1>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th><strong>Title</strong></th>
<th>Galaxy</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Training dataset:</strong></td>
<td>Nanopore MinION Sequencing of a Monkey Pox Virus (MPXV) from Spain 2022 oubreak. Data is publicly available at SRA with <a href="https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&amp;acc=ERR10297654&amp;display=metadata">ID ERR10297654</a>. <a href="https://pubmed.ncbi.nlm.nih.gov/38637500/">Paper</a></td>
</tr>
<tr>
<td><strong>Questions:</strong></td>
<td><ul><li>How Nanopore reads are differently assembled from Illumina?</li></ul></td>
</tr>
<tr>
<td><strong>Objectives</strong>:</td>
<td><ul><li>Understand the concept of assembly</li><li>Learn how to interpret assembly quality control metrics</li></ul></td>
</tr>
<tr>
<td><strong>Estimated time</strong>:</td>
<td>40 min</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<h2 id="1-description">1. Description</h2>
<p>Nanopore techology is a third generation sequencing technique which allows to get longer sequences, but with reduced sequence quality. Different technologies have different formats, qualities, and specific known biases which make the analysis different among them.
In this tutorial, we are going to see an example of how to assemble long reads from a Nanopore sequencing run.</p>
<h2 id="2-upload-data-to-galaxy">2. Upload data to galaxy</h2>
<h3 id="training-dataset">Training dataset</h3>
<ul>
<li>[SRA ID: ERR10297654](https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&amp;acc=ERR10297654&amp;display=metadata</li>
</ul>
<h3 id="create-new-history">Create new history</h3>
<ul>
<li>Click the <code>+</code> icon at the top of the history panel and create a new history with the name <code>nanopore assembly 101 tutorial</code> as explained <a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#2-galaxys-history">here</a></li>
</ul>
<h3 id="upload-data">Upload data</h3>
<ol>
<li>Look for <code>SRA</code> in the tool search bar and select <code>Faster Download and Extract Reads in FASTQ format from NCBI SRA</code></li>
<li>Accession = <code>ERR10297654</code></li>
<li>Execute</li>
</ol>
<p align="center"><img src="images/upload_sra_err.png" alt="upload_sra_err" width="900"></p>
<h3 id="load-reference-file-from-ncbi">Load reference file from NCBI</h3>
<ol>
<li>Search <code>NCBI</code> using the search toolbox and select <code>NCBI Accession Download Download sequences from GenBank/RefSeq by accession through the NCBI ENTREZ API</code></li>
<li>Select source for IDs &gt; Direct entry</li>
<li>ID List = NC_063383.1</li>
<li>Execute</li>
</ol>
<p align="center"><img src="images/load_ref_ncbi.png" alt="load_ref_ncbi" width="900"></p>
<h3 id="unhide-data">Unhide data</h3>
<p>Using SRA and NCBI API downloads data as hidden so we are going to unhidde this data as follows:</p>
<ol>
<li>Click on the strikethrough eye (Show hidden)</li>
<li>Select the strikethrough for ERR10297654 and NC_063383.1 datas.</li>
<li>Then select the location icon (show active)</li>
</ol>
<p><img src="images/unhide_1.png" alt="unhide_1" width="200"><img src="images/unhide_2.png" alt="unhide_2" width="200"></p>
<h3 id="mapping-with-minimap2">Mapping with Minimap2</h3>
<ol>
<li>Search <code>minimap2</code> using the search toolbox and select <code>Map with minimap2 A fast pairwise aligner for genomic and spliced nucleotide sequences</code></li>
<li>Will you select a reference genome from your history or use a built-in index?: Use a genome from history and built-in index
<ul>
<li>Select NC_063383.1</li>
</ul>
</li>
<li>Select fastq dataset: ERR10297654</li>
<li>Select a profile of preset options &gt; Oxford Nanopore Read to reference mapping (map-ont)</li>
<li>Click execute and wait.</li>
</ol>
<p align="center"><img src="images/minimap2_params.png" alt="minimap2_params" width="900"></p>
<h3 id="mapping-stats-with-samtools">Mapping stats with samtools</h3>
<ol>
<li>Search <code>flagstatst</code> using the search toolbox and select <code>Samtools flagstat tabulate descriptive stats for BAM datset</code></li>
<li>BAM File to report statistics of &gt; Select Minimap2 bam output</li>
<li>Click execute and wait.</li>
<li>Click in the 👁️ and see the bam stats.</li>
</ol>
<p align="center"><img src="images/samtools_flagstats.png" alt="samtools_flagstats" width="900"></p>
<details>
    <summary> Which is the mapping rate?</summary>
    </br>
    5.30%
</details>
<details>
    <summary>How many reads do we have in our dataset?</summary>
    21868
    </br>
</details>
<blockquote>
<p>This training history is available at: https://usegalaxy.eu/u/s.varona/h/nanopore-assembly-101-tutorial</p>
</blockquote>
<blockquote>
<p>Note: Nanopore data is known to have more error than short sequencing reads. This is why assembly post-processing is strongly recommended, usually using combined sequencing aproximation with both Nanopore and Illumina reads.</p>
</blockquote>

</body>
</html>
