<!DOCTYPE html>
<html>
<head>
<title>07_variant_calling_illumina.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>
<link rel="stylesheet" href="file:///Users/svarona/Repositories/galaxy_virologist_training/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="galaxy-for-virologist-training-exercise-6-illumina-variant-calling-101">Galaxy for virologist training Exercise 6: Illumina Variant Calling 101</h1>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th><strong>Title</strong></th>
<th>Galaxy</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Training dataset:</strong></td>
<td>PRJEB43037 - In August 2020, an outbreak of West Nile Virus affected 71 people with meningoencephalitis in Andalusia and 6 more cases in Extremadura (south-west of Spain), causing a total of eight deaths. The virus belonged to the lineage 1 and was relatively similar to previous outbreaks occurred in the Mediterranean region. Here, we present a detailed analysis of the outbreak, including an extensive phylogenetic study. This is one of the outbreak samples.</td>
</tr>
<tr>
<td><strong>Questions:</strong></td>
<td><ul><li>What is variant calling?</li><li>What is a vcf file?</li><li>How can I inspect a variant in a bam file to look for false positives?</li><li>How can I make a consensus genome based on a variant calling process?</li></ul></td>
</tr>
<tr>
<td><strong>Objectives</strong>:</td>
<td><ul><li>Understand variant calling concept</li><li>Learn how to interpret a vcf file</li><li>Learn how to make a reference consensus genome.</li><li>Learn how to visualize mapping and variant calling results</li></ul></td>
</tr>
<tr>
<td><strong>Estimated time</strong>:</td>
<td>2h</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<h2 id="1-description">1. Description</h2>
<p>After mapping, when we have a re-sequencing experiment, the next step usually comprises the variants calling step. Variant calling software tries to identify variants, positions that differ in our reads compared to a reference genome.
We may want to have a consensus genome as well, which is obtained by including the variants we just identified in the published reference genome.
We are going to address this type of analysis in this tutorial.</p>
<h2 id="2-upload-data-to-galaxy">2. Upload data to galaxy</h2>
<h3 id="training-dataset">Training dataset</h3>
<ul>
<li>Experiment info: PRJEB43037, WGS, Illumina MiSeq, paired-end</li>
<li>Fastq R1: <a href="https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_1.fastq.gz">ERR5310322_1</a> - url : <code>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_1.fastq.gz</code></li>
<li>Fastq R2: <a href="https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_2.fastq.gz">ERR5310322_2</a>  url : <code>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_2.fastq.gz</code></li>
<li>Reference genome NC_009942.1: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/875/385/GCF_000875385.1_ViralProj30293/GCF_000875385.1_ViralProj30293_genomic.fna.gz">fasta</a> -- <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/875/385/GCF_000875385.1_ViralProj30293/GCF_000875385.1_ViralProj30293_genomic.gff.gz">gff</a></li>
</ul>
<h3 id="create-new-history">Create new history</h3>
<ul>
<li>Click the <code>+</code> icon at the top of the history panel and create a new history with the name <code>variant calling 101 tutorial</code> as explained <a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/01_introduction_to_galaxy.md#2-galaxys-history">here</a></li>
</ul>
<h3 id="upload-data">Upload data</h3>
<p>Follow the same instructions <a href="https://github.com/BU-ISCIII/galaxy_virologist_training/blob/one_week_4day_format/exercises/03_mapping.md#2-upload-data-to-galaxy">here</a></p>
<pre class="hljs"><code><div>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR531/002/ERR5310322/ERR5310322_2.fastq.gz
https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/875/385/GCF_000875385.1_ViralProj30293/GCF_000875385.1_ViralProj30293_genomic.fna.gz
https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/875/385/GCF_000875385.1_ViralProj30293/GCF_000875385.1_ViralProj30293_genomic.gff.gz
</div></code></pre>
<p>Rename the data as follows:</p>
<ul>
<li><code>ERR5310322_1.fastq.gz</code> to <code>ERR5310322_1</code> with tag <code>#forward</code></li>
<li><code>ERR5310322_2.fastq.gz</code> to <code>ERR5310322_2</code> with tag <code>#reverse</code></li>
<li><code>GCF_000875385.1_ViralProj30293_genomic.fna.gz</code> to <code>NC_009942.1 fasta</code> with tag <code>#fastaref</code></li>
<li><code>GCF_000875385.1_ViralProj30293_genomic.gff.gz</code> to <code>NC_009942.1 gff</code> with tag <code>#gffref</code></li>
</ul>
<p align="center"><img src="images/rename_data.png" alt="rename_data" width="300"></p>
<h2 id="3-preprocess-our-reads">3. Preprocess our reads.</h2>
<p>Follow instructions <a href="02_quality.md#2-trimming">here</a></p>
<p>Then, fix fastp tags on the output data to be as follows:</p>
<p align="center"><img src="images/fastp_tags.png" alt="fastp_tags" width="300"></p>
<h2 id="4-map-trimmed-reads-against-the-reference-genome">4. Map trimmed reads against the reference genome.</h2>
<p>Follow:</p>
<ol>
<li>Is this single or paired library: paired.</li>
<li>FASTA/Q file #1 : fastp Read 1 output #forward</li>
<li>FASTA/Q file #2 : fastp Read 2 output #reverse</li>
<li>Will you select a reference genome from your history or use a built-in index? : Use a genome from the history and build index.</li>
<li>Do you want to use presets? : Very sensitive local. This setting will hugly affect the mapping results, depending on the dataset/experiment must be tweaked (read <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">bowtie2 manual</a>)</li>
<li>Save the bowtie2 mapping statistics to the history</li>
</ol>
<p align="center"><img src="images/bowtie2_vc1.png" alt="bowtie2_vc1" width="900"></p>
<p align="center"><img src="images/bowtie2_vc2.png" alt="bowtie2_vc2" width="900"></p>
<h2 id="5-variant-calling">5. Variant Calling</h2>
<h3 id="samtools-mpileup">Samtools mpileup</h3>
<ol>
<li>Search samtools mpileup in the search toolbox, scroll down and select <code>Samtools mpileup multi-way pileup of variants</code></li>
<li>Bam files: Bowtie2 bam file</li>
<li>Use reference: Use reference/genome from history. NC_009942.1.</li>
<li>Set advanced options: Advanced</li>
<li>Disable read-pair overlap detection: Yes</li>
<li>Disable BAQ (per-Base Alignment Quality), see below: Yes</li>
<li>Do not discard anomalous read pairs: Yes</li>
<li>max per-file depth; avoids excessive memory usage: 0</li>
<li>Minimum base quality for a base to be considered: 20</li>
<li>Click execute and wait.</li>
</ol>
<p align="center"><img src="images/samtools_mpileup_params1.png" alt="samtools mpileup" width="900"></p>
<p align="center"><img src="images/samtools_mpileup_params2.png" alt="samtools mpileup" width="900"></p>
<ol start="11">
<li>Click the :eye: icon on the history and inspect the mpileup output.</li>
</ol>
<h3 id="varscan">VarScan</h3>
<ol>
<li>Search <code>VarScan Mpileup</code> in the search toolbox.</li>
<li>Samtools pileup dataset: samtools mpileup output</li>
<li>Minimum read depth: 10</li>
<li>Minimum supporting reads: 5</li>
<li>Minimum base quality at a position to count a read: 20</li>
<li>Minimum variant allele frequency threshold: 0,75</li>
<li>Default p-value threshold for calling variants: 0,05</li>
<li>Click execute and wait</li>
</ol>
<p align="center"><img src="images/varscan_params1.png" alt="varscan" width="900"></p>
<ol>
<li>Click the :eye: icon and inspect the vcf file.</li>
</ol>
<h3 id="vcf-stats">VCF stats</h3>
<ol>
<li>Search <code>bcftools stats</code> in the search toolbox.</li>
<li>VCF/BCF Data: varscan vcf output.</li>
<li>Click execute and wait.</li>
<li>Click the :eye: icon and inspect the stats.</li>
</ol>
<details>
  <summary>How may variants do we have in our vcf file?</summary>
  </br>
  number of SNPs:	462
</details>
<h3 id="ivar-variants">Ivar variants</h3>
<ol>
<li>Search <code>ivar variants</code> in the search toolbox.</li>
<li>Select <code> ivar variants Call variants from aligned BAM file</code></li>
<li>Bam file: bowtie bam output</li>
<li>Reference:  NC_009942.1</li>
<li>Minimum quality score threshold to count base: 20</li>
<li>Minimum frequency threshold: 0.75</li>
<li>Output format: Both tabular and vcf</li>
<li>In VCF only output variants that PASS all filters &gt; Yes</li>
<li>Click execute and wait.</li>
</ol>
<p align="center"><img src="images/ivar_params1.png" alt="ivar" width="900"></p>
<h3 id="lofreq">Lofreq</h3>
<h4 id="insert-indel-qualities">Insert indel qualities</h4>
<ol>
<li>Search <code>Insert indel qualities</code> in the search toolbox. Select <code> Insert indel qualities into a BAM file</code></li>
<li>Reads: bowtie2 bam output.</li>
<li>Click execute and wait.</li>
</ol>
<p align="center"><img src="images/indel_qualities_params1.png" alt="lofreq indel qualities" width="500"></p>
<h4 id="call-variants">Call variants</h4>
<ol>
<li>Search <code>lofreq</code> in the search toolbox. Select Call variants with lofreq.</li>
<li>Input reads in BAM format: indel qualities bam output.</li>
<li>Choose the source for the reference genome: History. NC_009942.1</li>
<li>Types of variants to call: SNVs and INDELs</li>
<li>Variant calling parameters: Configure settings</li>
<li>Minimal coverage: 10</li>
<li>Minimum baseQ: 20</li>
<li>Minimum baseQ for alternate bases: 20</li>
<li>Click execute and wait.</li>
</ol>
<p align="center"><img src="images/lofreq_call_variants_params1.png" alt="lofreq call" width="900"></p>
<p align="center"><img src="images/lofreq_call_variants_params2.png" alt="lofreq call" width="900"></p>
<h2 id="compare-vcfs-among-callers">Compare vcfs among callers</h2>
<h3 id="visualize-datasets">Visualize datasets</h3>
<ol>
<li>Search <code>upSet diagram</code> in the search toolbox.</li>
<li>Select input files for which to produce intersections: select vcf from varscan, vcf from lofreq filter and vcf from ivar variants.</li>
<li>Click execute and wait.</li>
<li>Click the :eye: icon and check the diagram.</li>
</ol>
<details>
  <summary>How many variants differ among the vcfs?</summary>
  </br>
  There are up to 3 variants more in VarScan than iVar.
</details>
<h2 id="7-consensus-genome">7. Consensus genome</h2>
<h3 id="bcftools-consensus">Bcftools consensus</h3>
<ol>
<li>Search <code>bcftools consensus</code> in the search toolbox.</li>
<li>VCF/BCF Data: varscan vcf output.</li>
<li>Choose a reference genome: use genome/reference from history. Select NC_009942.1.</li>
<li>Click execute and wait.</li>
</ol>
<p align="center"><img src="images/bcftools_consensus_params.png" alt="varscan" width="500"></p>
<blockquote>
<p>Note: for this example we are not going to mask any position with low coverage, this will be addressed in the exercise 8, with a real example.</p>
</blockquote>
<h3 id="ivar-consensus">Ivar Consensus</h3>
<ol>
<li>Search <code>ivar consensus</code> in the search toolbox.</li>
<li>Bam file: bowtie bam output.</li>
<li>Use N instead of - for regions with less than minimum coverage: Yes</li>
</ol>
<p align="center"><img src="images/ivar_consensus_params1.png" alt="varscan" width="500"></p>
<blockquote>
<p>Here is the galaxy history for this exercise: <a href="https://usegalaxy.eu/u/smonzon/h/variant-calling-101-tutorial-1">https://usegalaxy.eu/u/smonzon/h/variant-calling-101-tutorial-1</a></p>
</blockquote>

</body>
</html>
